google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/AbstractBaseUnitOfWork.java

```diff
@@ -0,0 +1,159 @@
+/*
```

Comment:
Any tests we should be adding here? Or will be in future commits?

Reply:
The tests for both this abstract base class as well as much of the concrete subclasses are handled by the SQL based unit tests. I will add these through separate PRs, as they need the whole stack of connection and transactions to be present to work.